File: programming/python/practice/quickSort.py
def partition(arr,l,r):
    pivot=arr[r]
    j=l-1
    for i in range(l,r):
        if(arr[i]<=pivot):
            j=j+1
            arr[j],arr[i]=arr[i],arr[j]
        arr[j+1],arr[r]=arr[r],arr[j+1]
        return (j+1)


def quickSort(arr,l,r):

    if(l<r):
        pi=partition(arr,l,r)
        quickSort(arr,l,pi-1)
        quickSort(arr,pi+1,r)


arr=[5,4,3,2,1]
quickSort(arr,0,4)
n=len(arr)
for i in range(n):
    print(arr[i])

File: programming/DS /c/sortings/quickSort.c
#include<stdio.h>
void swap(int *a,int *b){
    int temp=*a;
    *a=*b;
    *b=temp;

}
int partition(int arr[],int l,int r){
    int pivot=arr[r];
    int i=l-1;
    for(int j=l;j<=r-1;j++){
        if(arr[j]<pivot){
            i++;
            swap(&arr[i],&arr[j]);
        }
    }swap(&arr[i+1],&arr[r]);

        return i+1;
}
void quickSort(int arr[],int l,int r){
    if(l<r){

        int pi=partition(arr,l,r);
        quickSort(arr,l,pi-1);
        quickSort(arr,pi+1,r);
    }
}
int main(){
    int arr[100];
    int n;
    scanf("%d",&n);
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }
    quickSort(arr,0,n-1);
    for(int i=0;i<n;i++){
        printf("%d",arr[i]);
    }

}